intermediate-python-testing.md

```diff
@@ -195,34 +196,51 @@ def database(request):
 * `module` – Fixture is run once per test module
 * `function` – Fixture is run every time a test includes it
 
-Fixtures are function-scoped by default, but it would be silly to re-create a database for every test. Instead, use the `session` scope for fixtures that establish context that you aren’t going to change over the course of your testing, such as creating a database, initializing an application, or inserting immutable dummy data.
+**Fixtures are function-scoped by default.** You should use this default unless you have a compelling reason. For example, it would be silly to re-create a database for every test. It is appropriate to use the `session` scope for fixtures that establish context that you aren’t going to change over the course of your testing, such as creating a database, initializing an application, or inserting immutable dummy data.
+
+**Be cautious!** Changes made to broadly scoped fixtures persist for all other tests in that session or module. This can lead to confusing circumstances where you are unsure whether your test is failing or the fixture context you expect was changed by a previous test.
+
+To diagnose these unintended dependencies, run your tests in a random order with [`pytest-randomly`](https://pypi.python.org/pypi/pytest-randomly). Simply:
+
+```bash
+pip install pytest-randomly
+```
+
+Then run your tests as normal and behold! They will be shuffled, and you will be kept honest. (Toggle random functionality with the `--randomly-dont-reorganize` flag if you need a run in sequential order.)
 
-Be cautious: Changes made to broadly scoped fixtures persist for all other tests in that session or module. This can lead to confusing circumstances where you are unsure whether your test is failing or the context was changed by a previous test. If you define a fixture that creates a table or inserts data that you intend to alter in your test, it’s better to use a narrower scope.
+If you define a fixture that creates a table or inserts data that you intend to alter in your test, **use the `function` scope** and write a finalizer.
 
-Calling narrowly scoped fixtures that create tables or insert data multiples times leads to a cluttered test database. This brings us to finalizers (see the `drop_database` definition in the example above). A finalizer is a method, defined within a fixture and decorated with `@request.addfinalizer`, that is run when that fixture falls out of scope. In effect, finalizers should undo your fixture. If you insert data, remove the data; if you create a table, delete the table; and so on.
+A finalizer is a method, defined within a fixture and decorated with `@request.addfinalizer`, that is run when that fixture falls out of scope. In effect, finalizers should undo your fixture. If you insert data, remove the data; if you create a table, delete the table; and so on. Let's add a finalizer that drops the database to the `database` fixture from before.
```

Comment:
I had to read this last sentence a couple times to glean its meaning. Either link to the above-referenced database fixture...or simplify the sentence (and assume that your reader can grasp the interconnectedness of the examples): "Let's add a finalizer that drops a database."